acceptable/responses.py

```diff
@@ -0,0 +1,75 @@
+import responses
+
+
+class ResponsesManager(object):
+    """The responses library is used to add mock behaviour into the requests
+    library. 
+    
+    It does this using the RequestsMock class, however only one of 
+    these can be active at a time. Attempting to start a new RequestsMock
+    will remove any others hooked into requests.
+    
+    We use an instance of this class to manage use of the RequestsMock 
+    instance `responses.mock`. This allows us to start, stop and reset 
+    the it at the right time.
+    """
+    def __init__(self):
+        self._attached = 0
+
+    def attach(self):
+        if self._attached == 0:
+            responses.mock.start()
+        self._attached += 1
+
+    def detach(self):
+        self._attached -= 1
+        assert self._attached >= 0
+        if self._attached == 0:
+            responses.mock.stop()
+            responses.mock.reset()
+
+    def attach_callback(self, methods, url, callback):
+        for method in methods:
+            responses.mock.add_callback(method, url, callback)
+        self.attach()
+
+    def detach_callback(self, methods, url, callback):
+        for method in methods:
+            responses.mock.remove(method, url)
+        self.detach()
+
+
+responses_manager = ResponsesManager()
+del ResponsesManager
```

Comment:
?!

Reply:
it is a bit weird, but I think it will reduce confusion. I can't think of any reason to want to create another ResponseManager.